crates/trie/src/node_iter.rs

```diff
@@ -6,28 +6,39 @@ use crate::{
 };
 use reth_primitives::{trie::Nibbles, Account, StorageEntry, B256, U256};
 
+/// Represents a branch node in the trie.
 #[derive(Debug)]
 pub(crate) struct TrieBranchNode {
+    /// The key associated with the node.
     pub(crate) key: Nibbles,
+    /// The value associated with the node.
     pub(crate) value: B256,
+    /// Indicates whether children are in the trie.
     pub(crate) children_are_in_trie: bool,
 }
 
 impl TrieBranchNode {
+    /// Creates a new `TrieBranchNode`.
     pub(crate) fn new(key: Nibbles, value: B256, children_are_in_trie: bool) -> Self {
         Self { key, value, children_are_in_trie }
     }
 }
 
+/// Represents a node within an account.
 #[derive(Debug)]
 pub(crate) enum AccountNode {
+    /// Branch node within the account.
     Branch(TrieBranchNode),
+    /// Leaf node within the account.
     Leaf(B256, Account),
 }
 
+/// Represents a node within storage.
 #[derive(Debug)]
 pub(crate) enum StorageNode {
+    /// Branch node within storage.
     Branch(TrieBranchNode),
+    /// Leaf node within storage.
     Leaf(B256, U256),
 }
```

Comment:
`within` is not quite right semantically, better to say `variant of`. e.g. for AccountNode `Represents a variant of an account node`

Reply:
Thanks for the clarification, just updated